synapse/replication/http/send_event.py

```diff
@@ -138,6 +140,16 @@ async def _handle_request(  # type: ignore[override]
             "Got event to send with ID: %s into room: %s", event.event_id, event.room_id
         )
 
+        if event.type == EventTypes.Member and event.membership == Membership.JOIN:
+            (
+                current_membership,
+                _,
+            ) = await self.store.get_local_current_membership_for_user_in_room(
+                event.state_key, event.room_id
+            )
+            if current_membership != Membership.JOIN:
+                self._notifier.notify_user_joined_room(event.event_id, event.room_id)
```

Comment:
I'm wondering if it makes more sense to call this at:

https://github.com/matrix-org/synapse/blob/757bc0caefa596e747278b3bcf4269ec50ffc759/synapse/handlers/message.py#L1523:L1523

which gets called on the right worker. That way the check in the `room_member.py` can always have `update=False`?

Reply:
I've made this change, but now I'm having real trouble fixing up the test case :/